src/flexalloc_mm.c

```diff
@@ -727,38 +708,74 @@ fla_base_sync(struct flexalloc *fs)
 }
 
 void
-fla_print_pool_entry(struct flexalloc *fs, struct fla_pool *pool)
+fla_print_geo(struct flexalloc *fs)
+{
+  struct fla_geo * geo = &(fs->geo);
+  fprintf(stderr, "Flexalloc Geometry: \n");
+  fprintf(stderr, "|  LBAs: %"PRIu64"\n", geo->nlb);
+  fprintf(stderr, "|  LBA width: %"PRIu32"B\n", geo->lb_nbytes);
+  fprintf(stderr, "|  md blocks: %"PRIu32"\n", geo->md_nlb);
+  fprintf(stderr, "|  pool segment:\n");
+  fprintf(stderr, "|    * pools: %"PRIu32"\n", geo->npools);
+  fprintf(stderr, "|    * freelist blocks: %"PRIu32"\n", geo->pool_sgmt.freelist_nlb);
+  fprintf(stderr, "|    * htbl blocks: %"PRIu32"\n", geo->pool_sgmt.htbl_nlb);
+  fprintf(stderr, "|    * entry blocks: %"PRIu32"\n", geo->pool_sgmt.entries_nlb);
+  fprintf(stderr, "|    * total blocks: %"PRIu32"\n", fla_geo_pool_sgmt_nblocks(&geo->pool_sgmt));
+  fprintf(stderr, "|  slab segment:\n");
+  fprintf(stderr, "|    * slabs: %"PRIu32"\n", geo->nslabs);
+  fprintf(stderr, "|    * total blocks: %"PRIu32"\n", geo->slab_sgmt.slab_sgmt_nlb);
+  fprintf(stderr, "\n");
+}
+
+void
+fla_print_pool_entries(struct flexalloc *fs)
 {
   int err = 0;
   struct fla_slab_header * curr_slab;
-  struct fla_pool_entry * pool_entry = &fs->pools.entries[pool->ndx];
-  uint32_t * heads[3] = {&pool_entry->empty_slabs, &pool_entry->full_slabs, &pool_entry->partial_slabs};
+  uint32_t * heads[3];
```

Comment:
heads seems a bit ambiguous

Reply:
Can we not use a pointer here?

Reply:
Sure. this was some clean up that I forgot to do. good catch.